### DISCUSSION TITLE
Classic and Modern presets for default settings

### DISCUSSION THREAD

**Discussion by ketsuban** [OpenTTD/OpenTTD#11245] [General]
I feel like if OpenTTD was a game being developed from scratch nowadays nobody would even suggest cargo distribution be disabled by default; this state of affairs only exists so OpenTTD can provide compatibility with old saves. As such, I wonder whether it would be possible to provide a couple different presets: Classic would be what the game currently comes with, emphasising compatibility rather than enabling new features, while Modern would provide sensible defaults for new players who don't care about compatibility with an ancient DOS game.

[Previous discussion I found of changing the default settings](https://github.com/OpenTTD/OpenTTD/discussions/8393) was focused on relatively small changes, so there's no appreciable overlap.

Is this a good idea? Are there any other settings which are currently disabled by default which would probably be enabled if OpenTTD were a new game being developed today (and which as a result would be candidates for a Modern preset)?

**Comment by 2TallTyler** [comment]
Cargo Distribution could be enabled by default without any impact on old saves. It just needs someone to open a PR proposing the change and explaining why. For what it's worth, I agree.

**Comment by zephyris** [comment]
Sounds like a decent idea to me too. Easy enough to suggest setting presets which are "As retro as possible", "TTDPatch-like", and "Classic". The contentious point would be what constitutes ideal modern preset - to me I agree that would include cargodist.

**Comment by 2TallTyler** [comment]
I'm not convinced that setting presets are needed. Don't most of us pick our favorite settings and leave them alone? (Leaving defaults mostly for new players.)

**Comment by JGRennison** [comment]
Cargodist isn't as simple as on/off.
I don't see that there is as clear a consensus on things like symmetric vs asymmetric for example.
The various tunables and their defaults may need further thought if it is to be on by default for some(?) cargo types.

**Reply by ketsuban** [reply]
Anecdotally, "symmetric for passengers and mail, asymmetric for everything else" seems like the rough consensus I've seen, but if a PR emerges to enable cargo distribution by default and you want to bikeshed about it I won't stop you. It seems to me the only reason there's "no clear consensus" is because so many people don't even realise cargo distribution is a *thing*—it's off by default, and turning it on requires diving into the settings tree (if you'll pardon the mixed metaphors).

**Comment by TrueBrain** [comment]
(this is in reply to 3 replies, so I am just making a new reply! :D )

Although changing a default setting is one thing, having a set of presets to choose from goes a bit wider than you might think. I once wrote some words about that:

https://gist.github.com/TrueBrain/976cab3b6c88f208636653191ba5c420

It is not only about what settings to pick, but also about what NewGRFs to load, what GSes, etc. It should, in my opinion, be a total package. Which also directly means you should be able to upload these presets to BaNaNaS, and on startup you should be presented with some built-in ones, and a quick way to fetch a list from BaNaNaS.

This resolves the whole discussion about "what cargodist settings exactly" etc etc. As people can just upload their own preset, so others can use them.

Talk about these things we had enough .. it mostly needs someone actually doing it. It is also not really complicated; but it does take effort. So we are mostly waiting for someone to come by to actually do it, instead of talk about it :)

**Reply by TrueBrain** [reply]
![108423928-c859de80-7238-11eb-99e0-20e5ffe00474](https://github.com/OpenTTD/OpenTTD/assets/1663690/21d4e4b2-9d93-4d6b-8a21-a060c1367410)

I mockup I once made for this interface.
